File: 서인/CH04/5_k번째_큰_수.js
function solution(n, k, card) {
  let answer;
  card.sort((a, b) => b - a);
  console.log(card);

  let s = new Set();
  let m = card.length;
  for (let i = 0; i < m; i++) {
    for (let j = i + 1; j < m; j++) {
      for (let k = j + 1; k < m; k++) {
        s.add(card[i] + card[j] + card[k]);
      }
    }
  }

  let arr = Array.from(s);
  answer = arr[k - 1];

  return answer;
}

let arr = [13, 15, 34, 23, 45, 65, 33, 11, 26, 42];
console.log(solution(10, 3, arr));

File: 종복/Java/Hash/5. K번째 큰수.java
import java.util.*;
class Main {	
	public int solution(int[] arr, int n, int k){
	    
	    int answer = -1; 
	    
	    TreeSet<Integer> set = new TreeSet<>(Collections.reverseOrder());
		
		for(int i=0; i<n; i++){
		    for(int j=i+1; j<n; j++){
		        for(int l=j+1; l<n; l++){
		            int sum = arr[i]+arr[j]+arr[l];
		            set.add(sum);
		        }
		    }
		}
		int order = 1; 
		
        for(int num: set){
            if(order == k){
                answer = num;
                break;
            }
            order++;
        }
      
		return answer;
	}

	public static void main(String[] args){
		Main T = new Main();
		Scanner kb = new Scanner(System.in);
		int n=kb.nextInt();
		int k=kb.nextInt();
		int[] arr=new int[n];
		for(int i=0; i<n; i++){
			arr[i]=kb.nextInt();
		}
		System.out.println(T.solution(arr, n, k));
	}
}
